### PR TITLE
Dev emergency api

### LINKED ISSUE #136: Emergency support backend API
## Context 

An API is needed to fetch data from the database and send it to the front end app

## Problem 
The backend does not have an API for this
## Solution 

- [x] Create Emergency support mini app 
- [x] Create Api
- [x] Connect Api with database schema

### LINKED ISSUE #137: Emergency support populate database api
## Context 

An API is needed to populate the database with emergency support info, this will be used by admins

## Problem 

The backend does not have an API for this

## Solution 

- [ ] Create Api for populating database
- [ ] Connect Api with database schema

### LINKED ISSUE #162: Documentation for Emergency support API
## Context
Emergency support API is used to fetch list of emergency suppports

## Problem
The Technical documentation of the API is needed 

## Solution 

- [ ] Create Documentation about how to setup the API (Overview of implementation details)
- [ ] Create Documentation about how to use the API (How to call the API )

### PR DESCRIPTION
**Pull request by jyaneshseeam**
# Pull Request name

Dev emergency API ( backend)

## Source 
Closes #136 
Closes #137 
Closes #162 

## Description 

Emergency support API, that sends a list of emergency supports available to front-end

### Summary of PR  

Emergency support API that returns a list of emergency supports + populating database functionality

### Purpose 
Populating emergency database functionality + Emergency list returned

## Review and Testing Steps
- [ ] you can jump directly to pr #161 and just check if all tests pass, if you want to play around with the API follow the steps below
- [ ] Pull this branch
- [ ] Run  ``` cd app\server ```
- [ ] create a .env file and get the values from the shared doc from https://docs.google.com/document/d/1sprMBncWt9RDJSFmwBZRFc_Gg_Zg0lG9kvmOKvTmca8/edit
- [ ]  run ``` npm install```
- [ ] run ``` docker build -t server . ```
- [ ] run ``` docker run -dp 8080:8080 server ```
- [ ] open request.http file
- [ ] POST http://localhost:8080/emergency/create with example in request.http   to create new emergency support
- [ ] GET http://localhost:8080/emergency/list to get list of emergency supports

## Additional Information (If applicable)
Check app\documentation\emergency_support.md for documentation of api
### Notes
you can use a tool like POSTMAN or a tool like Rest client

Copy and paste the request from request.http for Postman , or perform send request directly from that file for RestClient

### CODE REVIEW

**Comment by sontushkpaul** [comment]
@jyaneshseeam can we serve the emergency support list based on the user's geo location? Because the support list will vary in different location?

**Comment by jyaneshseeam** [comment]
> @jyaneshseeam can we serve the emergency support list based on the user's geo location? Because the support list will vary in different location?

Yes it's possible to do that, I just need to modify schema and add some additional Queries

**Comment by sontushkpaul** [comment]
> > @jyaneshseeam can we serve the emergency support list based on the user's geo location? Because the support list will vary in different location?
> 
> Yes it's possible to do that, I just need to modify schema and add some additional Queries

Great

**Comment by sontushkpaul** [comment]
> > > @jyaneshseeam can we serve the emergency support list based on the user's geo location? Because the support list will vary in different location?
> > 
> > 
> > Yes it's possible to do that, I just need to modify schema and add some additional Queries
> 
> Great

Are you planning to implement the location based support list in near future?

**Comment by jyaneshseeam** [comment]
> 

yeah im gonna do that this week itself push it to this pr

**Review by DemiAdisa — APPROVED** [review]
Works like a charm, great job

**Comment by DemiAdisa** [comment]
> Works like a charm, great job
I would suggest doing the location update in a different issue👌

**Comment by jyaneshseeam** [comment]
@DemiAdisa  yeah it would make things easier for other devs too to access the code here once it's merged to dev, I will open up new issue for the location based emergency api

**Comment by jyaneshseeam** [comment]
@sontushkpaul  admin endpoint added for "/users"

endpoint "/userInfo" now returns info specific to logged in user

File: app/server/account_management/controllers/user_management_controller.js [hunk 1]
```diff
@@ -78,6 +80,21 @@ const getUserList = async (req, res) => {
...
+ * This is for testing purposes or admin purposes.
+ */
+ const getUserInfo = async (req, res) => {
+    try {
+        const result = await model.getUserInfo(req.user);
+        res.status(200);
+        res.json(result);
+    } catch (err) {
+        res.status(500);
+        res.send(err.meassage);
```
Comment (ja-chisholm) [inline app/server/account_management/controllers/user_management_controller.js]: I think this is a typo

File: app/server/account_management/controllers/user_management_controller.js [hunk 2]
```diff
@@ -140,10 +157,76 @@ const getUserList = async (req, res) => {
...
+            if (result.status == 200) {
+                res.status(200);
+                res.json(result.message);
+            } else if (result.status == 401) {
+                res.status(401);
+                res.json(result.message);
+            }
+    } catch (err) {
+        res.status(500);
+        res.json(err.meassage);
```
Comment (ja-chisholm) [inline app/server/account_management/controllers/user_management_controller.js]: same here

File: app/server/account_management/controllers/user_management_controller.js [hunk 3]
```diff
@@ -140,10 +157,76 @@ const getUserList = async (req, res) => {
...
+            res.status(200);
+            res.json(result.message);
+        } else if (result.status == 401) {
+            res.status(401);
+            res.json(result.message);
+        }
+        
+    } catch (err) {
+        res.status(500);
+        res.json(err.meassage);
```
Comment (ja-chisholm) [inline app/server/account_management/controllers/user_management_controller.js]: same here

File: app/server/emergency_app/controllers/emergency_controller.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+ * This will return list of emergency support
+ */
+ const getEmergencyList = async (req, res) => {
+    try {
+        const result = await model.getEmergencyList();
+        res.status(200);
+        res.json(result);
+    } catch (err) {
+        res.status(500);
+        res.send(err.meassage);
```
Comment (ja-chisholm) [inline app/server/emergency_app/controllers/emergency_controller.js:18]: same here

Comment (jyaneshseeam) [reply app/server/emergency_app/controllers/emergency_controller.js:18]: Ohh yeah it's obviously a typo that got carried over, will correct it asap

**Comment by jyaneshseeam** [comment]
@ja-chisholm  typos corrected now, please approve

**Review by ja-chisholm — APPROVED** [review]

**Pull request merged**